coriolis/tests/osmorphing/test_debian.py

```diff
@@ -0,0 +1,297 @@
+        mock_grub2_cfg_editor.return_value.append_to_option.assert_has_calls(
+            [
+                mock.call(
+                    "GRUB_CMDLINE_LINUX_DEFAULT",
+                    {"opt_type": "key_val", "opt_key": "net.ifnames",
+                     "opt_val": 0}),
+                mock.call(
+                    "GRUB_CMDLINE_LINUX_DEFAULT",
+                    {"opt_type": "key_val", "opt_key": "biosdevname",
+                     "opt_val": 0}),
+                mock.call(
+                    "GRUB_CMDLINE_LINUX",
+                    {"opt_type": "key_val", "opt_key": "net.ifnames",
+                     "opt_val": 0}),
+                mock.call(
+                    "GRUB_CMDLINE_LINUX",
+                    {"opt_type": "key_val", "opt_key": "biosdevname",
+                     "opt_val": 0})
+            ]
+        )
+        mock_read_file.assert_called_once_with(
+            self.os_root_dir + '/etc/default/grub')
+        mock_write_file_sudo.assert_called_once_with(
+            "etc/default/grub", mock_grub2_cfg_editor.return_value.dump())
+        mock_exec_cmd_chroot.assert_called_once_with("/usr/sbin/update-grub")
+
+    @mock.patch('coriolis.utils.Grub2ConfigEditor')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_write_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_read_file')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_test_path')
+    def test_disable_predictable_nic_names_no_test_path(
+            self, mock_test_path, mock_read_file,
+            mock_write_file_sudo, mock_exec_cmd_chroot, mock_grub2_cfg_editor):
+
+        mock_test_path.return_value = False
+
+        self.morpher.disable_predictable_nic_names()
+
+        mock_read_file.assert_not_called()
+        mock_write_file_sudo.assert_not_called()
+        mock_exec_cmd_chroot.assert_not_called()
+        mock_grub2_cfg_editor.assert_not_called()
+
+    def test_get_update_grub2_command(self):
+        result = self.morpher.get_update_grub2_command()
+        self.assertEqual(result, "update-grub")
+
+    @mock.patch('coriolis.osmorphing.debian.StringIO')
+    def test__compose_interfaces_config(self, mock_stringio):
+        mock_stringio_instance = mock_stringio.return_value
+        result = self.morpher._compose_interfaces_config(self.nics_info)
+
+        mock_stringio_instance.write.assert_has_calls([
+            mock.call(debian.LO_NIC_TPL),
+            mock.call("\n\n"),
+            mock.call(debian.INTERFACES_NIC_TPL % {"device_name": "eth0"}),
+            mock.call("\n\n"),
+            mock.call(debian.INTERFACES_NIC_TPL % {"device_name": "eth1"}),
+            mock.call("\n\n")
+        ])
+        mock_stringio_instance.seek.assert_called_once_with(0)
+
+        self.assertEqual(result, mock_stringio_instance.read.return_value)
+
+    @mock.patch('yaml.dump')
```

Comment:
You can write the unit test without mocking `yaml.dump`.

You can use `yaml.dump` here in the unit test to do the assertion (it should match the `yaml.dump` from the tested function return statement).